chunkator/__init__.py

```diff
@@ -11,9 +11,12 @@ def chunkator(queryset, chunk_size):
     length. We're saving at least a ``count()`` query on QuerySets, or a
     CPU-and-RAM-consuming ``len(queryset)`` query.
     """
-    pk = 0
+    pk = None
     while True:
-        page = queryset.filter(pk__gt=pk).order_by('pk')[:chunk_size]
+        queryset = queryset.order_by('pk')
```

Comment:
yep, what @rodo says. django-chunkator is not designed to do pagination over non-unique sorting fields.
I'm wondering if I can add an "order_by" argument to ``chunkator`` to allow custom sorts. But by no means it should be used with a regular field (name, title...) because:

* if the table is updated between query "n" and query "n+1", you may miss a record in your export,
* if you have duplicates, and you're unlucky, you end up with missed records.

Examples:

* keys : A A A B C D,
* chunk_size: 2

the first chunk is "A A", but the second is "B C" and the last is "D", because you make ``filter(pk__gt=pk)``. You miss the third A.

And BTW, you onboarding example is not relevant. I assume that taks count in a process rarely go over 10.

(btw, this project and code is public ; maybe it's not a good idea to talk about our private concerns)